File: app/src/main/java/com/app/missednotificationsreminder/ui/fragment/ReminderFragment.java
package com.app.missednotificationsreminder.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.app.missednotificationsreminder.binding.model.ReminderViewModel;
import com.app.missednotificationsreminder.binding.model.SettingsViewModel;
import com.app.missednotificationsreminder.databinding.ReminderViewBinding;
import com.app.missednotificationsreminder.ui.fragment.common.CommonFragmentWithViewModel;

import javax.inject.Inject;

import dagger.android.ContributesAndroidInjector;

/**
 * Fragment which displays interval settings view
 *
 * @author Eugene Popovich
 */
public class ReminderFragment extends CommonFragmentWithViewModel<ReminderViewModel> {

    @Inject
    ReminderViewModel model;
    @Inject SettingsViewModel parentModel;
    ReminderViewBinding mBinding;

    @Override public ReminderViewModel getModel() {
        return model;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        mBinding = ReminderViewBinding.inflate(inflater, container, false);
        return mBinding.getRoot();
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init(view, savedInstanceState);
    }

    private void init(View view, Bundle savedInstanceState) {
        mBinding.setModel(model);
        mBinding.setParentModel(parentModel);
    }

    @dagger.Module
    public static abstract class Module {
        @ContributesAndroidInjector
        abstract ReminderFragment contribute();
    }
}

File: app/src/main/java/com/app/missednotificationsreminder/settings/reminder/ReminderFragment.kt
package com.app.missednotificationsreminder.settings.reminder

import android.os.Bundle
import android.view.View
import androidx.fragment.app.viewModels
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.asLiveData
import com.app.missednotificationsreminder.R
import com.app.missednotificationsreminder.databinding.FragmentReminderBinding
import com.app.missednotificationsreminder.di.ViewModelKey
import com.app.missednotificationsreminder.settings.SettingsViewModel
import com.app.missednotificationsreminder.ui.fragment.common.CommonFragmentWithViewBinding
import dagger.Binds
import dagger.android.ContributesAndroidInjector
import dagger.multibindings.IntoMap
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.FlowPreview
import javax.inject.Inject

/**
 * Fragment which displays interval settings view
 */
@FlowPreview
@ExperimentalCoroutinesApi
class ReminderFragment : CommonFragmentWithViewBinding<FragmentReminderBinding>(R.layout.fragment_reminder) {
    @Inject
    lateinit var viewModelFactory: ViewModelProvider.Factory

    private val viewModel by viewModels<ReminderViewModel> { viewModelFactory }

    @Inject
    lateinit var parentModel: SettingsViewModel

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    private fun init() {
        viewDataBinding.apply {
            // Set the lifecycle owner to the lifecycle of the view
            lifecycleOwner = viewLifecycleOwner
            parentModel = this@ReminderFragment.parentModel
            viewModel = this@ReminderFragment.viewModel
            viewState = this@ReminderFragment.viewModel.viewState.asLiveData()
        }
    }

    @dagger.Module
    abstract class Module {
        @ContributesAndroidInjector
        abstract fun contribute(): ReminderFragment

        @Binds
        @IntoMap
        @ViewModelKey(ReminderViewModel::class)
        internal abstract fun bindViewModel(viewModel: ReminderViewModel): ViewModel
    }
}